src/dbnode/persist/fs/seek.go

```diff
@@ -143,32 +143,33 @@ type fileSetSeeker interface {
 
 func newSeeker(opts seekerOpts) fileSetSeeker {
 	return &seeker{
-		filePathPrefix: opts.filePathPrefix,
-		keepUnreadBuf:  opts.keepUnreadBuf,
-		bytesPool:      opts.bytesPool,
-		decoder:        msgpack.NewDecoder(opts.decodingOpts),
-		decodingOpts:   opts.decodingOpts,
-		opts:           opts,
+		opts: opts,
 	}
 }
 
 func (s *seeker) ConcurrentIDBloomFilter() *ManagedConcurrentBloomFilter {
 	return s.bloomFilter
 }
 
-func (s *seeker) Open(namespace ident.ID, shard uint32, blockStart time.Time) error {
+func (s *seeker) Open(
+	namespace ident.ID,
+	shard uint32,
+	blockStart time.Time,
+	resources ReusableSeekerResources,
+) error {
 	if s.isClone {
 		return errClonesShouldNotBeOpened
 	}
 
-	shardDir := ShardDataDirPath(s.filePathPrefix, namespace, shard)
-	var infoFd, indexFd, dataFd, digestFd, bloomFilterFd, summariesFd *os.File
+	shardDir := ShardDataDirPath(s.opts.filePathPrefix, namespace, shard)
```

Comment:
Hm do we still need the local var if right below we're setting `s.shardDir = shardDir`?

Reply:
good point